webapp/channels/src/packages/mattermost-redux/src/actions/emojis.ts

```diff
@@ -69,16 +72,52 @@ export function getCustomEmojiByName(name: string): ActionFunc {
 }
 
 export function getCustomEmojisByName(names: string[]): ActionFunc {
-    return async (dispatch: DispatchFunc) => {
+    return async (dispatch: DispatchFunc, getState: GetStateFunc) => {
         if (!names || names.length === 0) {
             return {data: true};
         }
 
-        const promises: Array<Promise<ActionResult|ActionResult[]>> = [];
-        names.forEach((name) => promises.push(dispatch(getCustomEmojiByName(name))));
+        // If necessary, split up the list of names into batches based on api4.GetEmojisByNamesMax on the server
+        const batchSize = 200;
 
-        await Promise.all(promises);
-        return {data: true};
+        const batches = [];
+        for (let i = 0; i < names.length; i += batchSize) {
+            batches.push(names.slice(i, i + batchSize));
+        }
```

Comment:
Wonder if we really want to send all the requests at once. Doesn't it slightly defeat the purpose of batching? And what about if we try to fetch 10000 emojis for whatever reason? 0/5